app-libs/exchange-oracle/src/coingecko.rs

```diff
@@ -50,21 +53,37 @@ lazy_static! {
 	]);
 }
 /// REST client to make requests to CoinGecko.
-pub struct CoinGeckoClient {
+pub struct CoinGeckoClient<OCallApi> {
 	client: RestClient<HttpClient>,
+	ocall_api: Arc<OCallApi>,
```

Comment:
The client needs the OCall-API to send the metrics updates to the untrusted service (where the metrics exporter lives)